jdaviz/app.py

```diff
@@ -1280,6 +1299,18 @@ def add_data(self, data, data_label=None, notify_done=True):
 
         self.data_collection[data_label] = data
 
+        # manage associated Data entries:
+        self._add_assoc_data_as_parent(data_label)
+        if parent is not None:
+            # Does the parent Data have a parent? If so, raise error:
+            parent_of_parent = self._get_assoc_data_parent(parent)
+            if parent_of_parent is not None:
+                raise NotImplementedError('Data associations are currently supported '
+                                          'between root layers (without parents) and their '
+                                          f'children, but the proposed parent "{parent}" has '
+                                          f'parent "{parent_of_parent}".')
+            self._set_assoc_data_as_child(data_label, new_parent_label=parent)
```

Comment:
do we also need to validate that `parent` exists in the data-collection?

Reply:
Sure thing, done in [55d2e68](https://github.com/spacetelescope/jdaviz/pull/2761/commits/55d2e688c44d5ef4a86d624a978619c300f9ff87).